akka-http2-support/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala

```diff
@@ -66,8 +68,12 @@ private[http2] object RequestParsing {
           rec(remainingHeaders.tail, method, scheme, authority, path, contentType, contentLength, headers)
 
         case (name, value) ⇒
-          // FIXME: parse to real headers
-          rec(remainingHeaders.tail, method, scheme, authority, path, contentType, contentLength, headers += RawHeader(name, value))
+          // FIXME: later modify by adding HttpHeaderParser.parseHttp2Header that would use (name, value) pair directly
+          //        or use a separate, simpler, parser for Http2
+          val concHeaderLine = name + ": " + value + "\r\nx"
```

Comment:
That is a bit ugly, but a by-product of how current parser works. Without `\r\n\x` (x being any additional byte) parsing will fail. See HttpHeaderParserSpec